diffsims/generators/simulation_generator.py

```diff
@@ -0,0 +1,428 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2024 The diffsims developers
+#
+# This file is part of diffsims.
+#
+# diffsims is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# diffsims is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with diffsims.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Kinematic Diffraction Simulation Generator."""
+
+from typing import Union, Sequence
+import numpy as np
+
+from orix.quaternion import Rotation
+from orix.crystal_map import Phase
+
+from diffsims.crystallography import ReciprocalLatticeVector, DiffractingVector
```

Comment:
Can the diffracting vector class be private?

You state in its docstring that it is not meant to be used by end-users. This tells me that the logic should be a simpler class or sets of functions to handle this particular use-case, instead of a public class with many caveats.

Reply:
If the user wants the original reciprocal lattice vectors, rotations, or rotated vectors, can they instead get `ReciprocalLatticeVector` or `Rotation` (preferably the more powerful `Orientation`)?